RecoBTag/DeepFlavour/plugins/sorting_modules.h

```diff
@@ -0,0 +1,88 @@
+#ifndef RecoBTag_DeepFlavour_sorting_modules_h
+#define RecoBTag_DeepFlavour_sorting_modules_h
+
+#include <algorithm>
+#include <cmath>
+#include <vector>
+
+namespace btagbtvdeep{
+
+/*
+ * the std::sort function needs a strict ordering.
+ * means if your_function(a,b) is true, than your_function(b,a) must be false for all a and b
+ */
+
+template <class T>
+class SortingClass{
+
+  public:
+
+    SortingClass():sortValA(0),sortValB(0),sortValC(0),t_(0) {}
+
+    SortingClass(const T& t, float sortA, float sortB=0, float sortC=0) :
+      t_(t), sortValA(sortA), sortValB(sortB), sortValC(sortC) {}
+
+    const T& get() const {return t_;}
+ 
+    enum compareResult{cmp_smaller,cmp_greater,cmp_invalid};
+
+    static inline bool isPhysValue(const float& val){
+        if (val!=val)return false;
+        if (std::isinf(val)) return false;
+        return true;
+    }
```

Comment:
can we use edm::isFinite here?

Reply:
I am checking the actual edm::isFinite implementation:
https://github.com/cms-sw/cmssw/blob/09c3fce6626f70fd04223e7dacebf0b485f73f54/FWCore/Utilities/interface/isFinite.h
The behaviour is not documented and it does some bitwise AND with a mask which seems non-trivial, does it check for null and inf as this function? If that is the case I guess we could use that implementation.

Reply:
nans and infinities (only) have all ones in the exponent field. The edm::isFinite checks for this.